File: lc/c/055.c
// Hash Table
bool isAnagram(char * s, char * t){
    int slen = strlen(s), tlen = strlen(t);
    if (slen != tlen)
        return false;

    int cnt[128] = {0};
    for (int i = 0; i < slen; ++i) {
        cnt[s[i]] += 1;
        cnt[t[i]] -= 1;
    }

    for (int i = 0; i < 128; ++i) {
        if (cnt[i] != 0)
            return false;
    }
    return true;
}

// Sorting
static int compare(void *a, void *b) {
    return *(char*)a - *(char*)b;
}

bool isAnagram(char * s, char * t){
    if (strlen(s) != strlen(t))
        return false;

    qsort(s, strlen(s), sizeof *s, compare);
    qsort(t, strlen(t), sizeof *t, compare);
    return strcmp(s, t) == 0 ? true : false;
}

File: lc/py/055.py
# Hash Table
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        x, y = Counter(s), Counter(t)
        return True if x == y else False

# Sorting
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        return ''.join(sorted(s)) == ''.join(sorted(t))
